[Darren Hoon] iP

darrenhoon opened the pull request:
### DukePro

“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html)
2. double click it.
3. add your tasks,
4. let it manage your tasks for you!

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainWindow.class, args);
    }
}
```

darrenhoon committed:
Testing git commit

darrenhoon committed:
Level-1

darrenhoon committed:
Level-1 done

darrenhoon committed:
Level-2 Done

darrenhoon committed:
Level-3 Done

darrenhoon committed:
Level-4 Done

darrenhoon committed:
A-TextUiTesting Done

darrenhoon committed:
Level-5 Error Handling Done

darrenhoon committed:
Level-6 Done

darrenhoon committed:
added java docs

darrenhoon committed:
Handled negative index error

darrenhoon committed:
Refactored code

darrenhoon committed:
added javadoc to messages

darrenhoon committed:
Removed unused enum

darrenhoon committed:
Refactor javadocs

darrenhoon committed:
Add Jackson JSON

darrenhoon committed:
Level-7 Done

darrenhoon committed:
Level-8 Done

darrenhoon committed:
Change database

darrenhoon committed:
Level-9 Done

darrenhoon committed:
Add info

darrenhoon committed:
Level-9 Done

darrenhoon committed:
Edit README

darrenhoon committed:
Add JavaDocs

darrenhoon committed:
Comply with Coding Standard

darrenhoon committed:
Merge branch-A-JavaDoc and branch-A-CodingStandard

darrenhoon committed:
A-MoreOOP

darrenhoon committed:
A-Jar

darrenhoon committed:
A-Packages

darrenhoon committed:
A-JUnit Done

darrenhoon committed:
Add header comment all files

darrenhoon committed:
Add JavaFX

darrenhoon committed:
Add JavaFX testing

darrenhoon committed:
Add JavaFX GUI:

darrenhoon committed:
Done A-Gradle

g4ryy commented on src/main/java/exception/DukeException.java:
```diff
@@ -0,0 +1,19 @@
+public class DukeException extends Exception {
+
+    /**
+     * Superclass of all exceptions that could be thrown by the Duke app.
```
Perhaps it would be better if the description describes the constructor rather than the class, since you have already done that above. I also noticed this issue in InvalidCommandException.java and TaskNumberException.java.

g4ryy commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,60 @@
+public class Parser {
+
+    /**
+     * Method to split a string into an array, removing spaces.
```
Perhaps it would be better if the sentence could start in the form "Splits a string into ...".

darrenhoon replied:
thanks for the feedback Gary!

g4ryy commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,205 @@
+
+        try {
+            if(!Files.exists(this.dir)) {
+                messages.displayText("You currently do not have a directory to save the database. " +
```
Perhaps breaking before the '+' operator would be better.

g4ryy commented on src/main/java/ui/FurtherAction.java:
```diff
@@ -0,0 +1,5 @@
+package ui;
+
```
Perhaps it would be good if you could include a Javadoc comment to describe what this enum is for.

g4ryy commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,136 @@
+
+    /**
+     * Displays message input with horizontal borderlines on top and bottom.
+     * @param input String message to be displayed to user
```
Perhaps you could end the description with a full stop.

g4ryy reviewed (commented):
Overall, I found your code really easy to read! There are only a few minor coding standard violations that needs to be fixed, but other than that everything looks good! 🙂

darrenhoon committed:
Done Level-10

darrenhoon committed:
Add header comment Main

darrenhoon committed:
Add header comment Launcher

darrenhoon committed:
Merge branch 'branch-Level-10'

darrenhoon committed:
Add more header comments

darrenhoon committed:
Merge branch 'branch-Level-10'

darrenhoon committed:
Done CheckStyle

darrenhoon committed:
Add header comments

darrenhoon committed:
Configure Gradle Done

darrenhoon committed:
Used gradle to generate the JAR file for the project.

* Project works with gradle test cases
* jar file is named ip.jar

Signed-off-by: darrenhoon <darrenhoon.ad@gmail.com>

darrenhoon committed:
Add spacing to Duke main method

As per the JavaDoc convention, there is to an empty line between the
description of the method and the parameters of the method.

- Duke main affected
- other .java files need to be amended too

darrenhoon committed:
Signed-off-by: darrenhoon <darrenhoon.ad@gmail.com>

Add spacing to header comment

As per JavaDoc's standard, there needs to be a spacing between the
description and the parameters and return values.

- Added lines to all files less duke/ and ui/
- ui/ file lines to be edited soon
- testing out commit --signoff command too

darrenhoon committed:
Add Assert, format ui/ per javadocs

darrenhoon committed:
Merge pull request #1 from darrenhoon/branch-A-Assertions

Add Assert, format ui/ per javadocs

darrenhoon committed:
Edit header comments

darrenhoon committed:
Merge branch 'master' of https://github.com/darrenhoon/ip

darrenhoon committed:
Improve code quality

The following quality improvements were made:
* refactored common code
* reduced methods to be mostly within 30 LOCs
* changed variable names to be more meaningful

Signed-off-by: darrenhoon <darrenhoon.ad@gmail.com>

darrenhoon committed:
Merge pull request #2 from darrenhoon/branch-A-CodeQuality

Improve code quality

darrenhoon committed:
Add Tag ArrayList to Task

darrenhoon committed:
Add basic-tag

darrenhoon committed:
Tag Todo Done

darrenhoon committed:
Deadline Tag Done

darrenhoon committed:
Event Tag Done

darrenhoon committed:
Gui tag added

darrenhoon committed:
Fix DialogBox height

darrenhoon committed:
Add style to GUI

darrenhoon committed:
Add CI

darrenhoon committed:
Add CI

darrenhoon committed:
Streams used

darrenhoon committed:
Remove magic literals

darrenhoon committed:
JavaDocs edit

darrenhoon committed:
Remove magic literals Parser

darrenhoon committed:
Add BetterGui

darrenhoon committed:
tag small removed

darrenhoon committed:
Update Ui.png docs/README.md

darrenhoon committed:
Set theme jekyll-theme-cayman

darrenhoon committed:
Update readme

darrenhoon committed:
Merge branch 'master' of https://github.com/darrenhoon/ip

darrenhoon committed:
Update UG

darrenhoon committed:
edit Readme

darrenhoon committed:
edit README

darrenhoon committed:
Rename Duke to FakeApp

darrenhoon committed:
Fix UG
